### ISSUE TITLE
Evaluation metric aucpr does not work for multiclass

### ISSUE THREAD

**Issue by lorenzwalthert** [dmlc/xgboost#5662]
I think it would be awesome if that would be supported. If not, we should document that it's not working for multiclass. Here is how it fails currently.

``` r
library(xgboost)
data(agaricus.train, package='xgboost')
data(agaricus.test, package='xgboost')

dtrain <- xgb.DMatrix(agaricus.train$data, label = agaricus.train$label + 1*sample(0:1, length(agaricus.train$label), replace = TRUE))
dtest <- xgb.DMatrix(agaricus.test$data, label = agaricus.test$label + 1*sample(0:1, length(agaricus.test$label), replace = TRUE))
watchlist <- list(train = dtrain, eval = dtest)

## merror works
param <- list(max_depth = 2, eta = 1, verbose = 0, nthread = 2, num_class = 3,
              objective = "multi:softprob", eval_metric = "merror")
bst <- xgb.train(param, dtrain, nrounds = 2, watchlist)
#> [1]  train-merror:0.493321   eval-merror:0.490379 
#> [2]  train-merror:0.486565   eval-merror:0.482930

## aucpr does not
param <- list(max_depth = 2, eta = 1, verbose = 0, nthread = 2, num_class = 3,
              objective = "multi:softprob", eval_metric = "aucpr")
bst <- xgb.train(param, dtrain, nrounds = 2, watchlist)
#> Error in xgb.iter.eval(bst$handle, watchlist, iteration - 1, feval): [18:53:59] amalgamation/../src/metric/rank_metric.cc:456: Check failed: preds.Size() == info.labels_.Size() (19539 vs. 6513) : label size predict size not match
#> Stack trace:
#>   [bt] (0) /home/walor/R/x86_64-pc-linux-gnu-library/3.6/xgboost/libs/xgboost.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7fbb2dded9ec]
#>   [bt] (1) /home/walor/R/x86_64-pc-linux-gnu-library/3.6/xgboost/libs/xgboost.so(+0x1840df) [0x7fbb2dec90df]
#>   [bt] (2) /home/walor/R/x86_64-pc-linux-gnu-library/3.6/xgboost/libs/xgboost.so(xgboost::LearnerImpl::EvalOneIter(int, std::vector<xgboost::DMatrix*, std::allocator<xgboost::DMatrix*> > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x439) [0x7fbb2deacab9]
#>   [bt] (3) /home/walor/R/x86_64-pc-linux-gnu-library/3.6/xgboost/libs/xgboost.so(XGBoosterEvalOneIter+0x135) [0x7fbb2de08fb5]
#>   [bt] (4) /home/walor/R/
```

<sup>Created on 2020-05-13 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by hcho3** [comment]
@lorenzwalthert Did you try `objective='multi:softmax'`?

**Comment by hcho3** [comment]
I've opened #5899 to document the limitation.

**Comment by hcho3** [comment]
Scikit-learn has an implementation for AUCROC that works with multi-class classification task: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html. I think this can be used with XGBoost.
